### DISCUSSION TITLE
[Unstable] Half of the crawlers are unable to board submarines from birth

### DISCUSSION THREAD

**Discussion by SomeRandomNoobKekeke** [FakeFishGames/Barotrauma#13755] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I was testing https://github.com/FakeFishGames/Barotrauma/discussions/13751 and found that it was a symptom of a bigger problem, so i made another report 

I was unable to reproduce it with mudraptors, husks, crawler husks, large crawlers ,threshers, hammerheadspawn
only with normal crawlers (and crawler hatchlings)

I don't understand why half is broken and half is normal, this seems random
They don't get stuck in stable

https://github.com/FakeFishGames/Barotrauma/assets/122838333/c8b30b66-405e-4a35-a051-1848e398105a

https://github.com/FakeFishGames/Barotrauma/assets/122838333/c9d9a0c6-6682-47c5-9cdb-e4a1257fa071

### Test 1
open some sub in editor, lock it and explode it
put some human bait inside and godmode it
spawn some crawlers outside and watch them get stuck in the walls

### Test 2
open some sub in editor, lock it and explode it
spawn crawler, control it, try to board the sub
kill it, repeat

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.4.0.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by AzureNocte** [comment]
I think this has something to do with 50% latchonto subs chance of crawler:
`    <chooserandom>
      <latchonto attachtowalls="true" attachtosub="true" mindeattachspeed="5.0" maxdeattachspeed="8.0" damageondetach="30" detachstun="5.0" attachlimb="Head" localattachpos="40,10" offset="50" attachlimbrotation="100" />
      <latchonto attachtowalls="true" attachtosub="false" mindeattachspeed="5.0" maxdeattachspeed="8.0" damageondetach="30" detachstun="5.0" attachlimb="Head" localattachpos="40,10" offset="50" attachlimbrotation="100" />
    </chooserandom>`

As far as I know, only crawler has this percentage based latch onto subs behavior.

**Reply by SomeRandomNoobKekeke** [reply]
I removed that lines from Crawler.xml and they stopped getting stuck
indeed it seems to be caused by latchonto

**Reply by SomeRandomNoobKekeke** [reply]
and this is why crawler_large is unaffected 

```
<Charactervariant inherit="Crawler" speciesname="Crawler_large" speciestranslationoverride="Crawler" usebosshealthbar="false">
  <health vitality="160" poisonvulnerability="1"/>
  <ai combatstrength="160" avoidgunfire="False">
    <!--Disables latching-->
    <chooserandom>
      <clear/>
    </chooserandom>
  </ai>
  <ragdoll scalemultiplier="1.3"/>
  <attack damagemultiplier="1.3" rangemultiplier="1.3" impactmultiplier="1.3" />
</Charactervariant>
```

**Comment by Regalis11** [comment]
Thank you for the report!

It is intentional that 50% of the crawlers latch on the the sub's walls, but there seems to be something off here: they should be able to get past broken walls. [I created an issue ticket about this](https://github.com/FakeFishGames/Barotrauma/issues/13756), we'll look into it!
